cfme/tests/intelligence/reports/test_reports.py

```diff
@@ -208,6 +217,49 @@ def timezone(appliance):
     appliance.user.my_settings.visual.timezone = current_timezone
 
 
+@pytest.fixture(scope="module")
+def setup_replicated_multiple_appliances_with_providers(
+    temp_appliances_unconfig_modscope_rhevm, replicated_appliances_modscope
+):
+    # In this case, first appliance from the 2 appliances spawned by
+    # `temp_appliances_unconfig_modscope_rhevm` is used by
+    # global_appliance in `replicated_appliances_modscope`
+    second_remote_appliance = temp_appliances_unconfig_modscope_rhevm[1]
+    remote_appliance, global_appliance = replicated_appliances_modscope
+    logger.info("Starting appliance replication configuration.")
+    second_remote_appliance.configure(region=89, key_address=remote_appliance.hostname)
+    second_remote_appliance.set_pglogical_replication(replication_type=":remote")
```

Comment:
Instead of setting up some of the appliances in `replicated_appliances_modscope` and some of them here, I think it would make more sense to have all of the db configuration and replication configuration in one fixture. The fixtures in `cfme/fixtures/multi_region.py` could be used here; `multi_region_cluster` can pass an indirect parameter to `temp_appliances_unconfig_modscope_rhevm` to specify the total number of appliances, and `setup_multi_region_cluster` takes care of configuring all of them. There is also a `setup_remote_provider` fixture that could be copied or modified to set up providers on multiple remote appliances.

Reply:
This was awesome! Thanks for the suggestions @tpapaioa, let me know how it looks now.